ibis/sql/alchemy.py

```diff
@@ -46,6 +54,12 @@
     dt.Int64: sa.BigInteger,
 }
 
+if GEO:
+    # Geospatial types
+    _ibis_type_to_sqla.update({
+        dt.GeoSpatial: ga.Geometry
```

Comment:
Unsure just now how to handle `Geometry` vs `Geography` types.

Reply:
at this time, each geospatial object/type (point, linestring, etc) has additional information: srid, geotype (geometry or geography) and nullable: https://github.com/ibis-project/ibis/blob/c2323b8dfd7b56db821426513c379de38203b332/ibis/expr/datatypes.py#L585

for example: `l_point = ibis.literal((0, 0), type='point;4326:geometry')`

Reply:
Geoalchemy has two different classes (`Geometry` and `Geography`), each of which can be one of the geometric subtypes. So mapping an `ibis.GeoSpatial` to one of those based on the type alone does not seem doable. Or do I misunderstand what you mean?

Reply:
I see ... not sure how you can handle this in this case .. I need to check geoalchemy ... `mapd` maps in inverse order `point, linestring, polygon ...` and each one can be a geometry or a geography ... so there is no map directly to `dt.GeoSpatial`

 if you really need these to types you can create here:

https://github.com/ibis-project/ibis/blob/c2323b8dfd7b56db821426513c379de38203b332/ibis/expr/types.py#L843
https://github.com/ibis-project/ibis/blob/c2323b8dfd7b56db821426513c379de38203b332/ibis/expr/datatypes.py#L585

maybe these two classes could be based on GeoSpatial and you could add a new parameter to store if it is a point, string, etc

does it make sense?

Reply:
another option is don't update `_ibis_type_to_sqla` and treat the type directly on: https://github.com/ibis-project/ibis/blob/dc1bbacff88a3dc0324f747b74b8c0b774985816/ibis/sql/alchemy.py#L64

so you can just check the type and return something like `GEOMETRY('POINT')` (for an ibis geometry point) if there is no information about geotypes ( geometry or geography) ... so you could use `GEOMETRY` as default.